src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+    // itself. If that is the case, the road is obviously not a sliproad.
+    if (canBeTargetOfSliproad(first->intersection) && at != second->node)
+        return boost::make_optional(std::size_t{2});
+
+    if (canBeTargetOfSliproad(second->intersection) && at != first->node)
+        return boost::make_optional(std::size_t{1});
+
+    return boost::none;
+}
+
+bool SliproadHandler::nextIntersectionIsTooFarAway(const NodeID start, const EdgeID onto) const
+{
+    BOOST_ASSERT(start != SPECIAL_NODEID);
+    BOOST_ASSERT(onto != SPECIAL_EDGEID);
+
+    const auto &extractor = intersection_generator.GetCoordinateExtractor();
+
+    // TODO: refactor, could be useful for other scenarios, too
+
+    // Base max distance threshold on the current road class we're on
+    const auto &data = node_based_graph.GetEdgeData(onto);
+    const auto threshold = scaledThresholdByRoadClass(MAX_SLIPROAD_THRESHOLD, // <- scales down
+                                                      data.road_classification);
+
+    struct NextIntersectionDistanceAccumulator
+    {
+        NextIntersectionDistanceAccumulator(
+            const extractor::guidance::CoordinateExtractor &extractor_,
+            const util::NodeBasedDynamicGraph &graph_,
+            const double threshold)
+            : extractor{extractor_}, graph{graph_}, threshold{threshold}, too_far_away{false},
+              distance{0}
+        {
+        }
+
+        bool terminate()
+        {
+            if (distance > threshold)
+            {
+                too_far_away = true;
+                return true;
+            }
+
+            return false;
+        }
+
+        void update(const NodeID start, const EdgeID onto, const NodeID)
+        {
+            using namespace util::coordinate_calculation;
+
+            const auto coords = extractor.GetForwardCoordinatesAlongRoad(start, onto);
+            distance += getLength(coords, &haversineDistance);
+        }
+
+        const extractor::guidance::CoordinateExtractor &extractor;
+        const util::NodeBasedDynamicGraph &graph;
+        const double threshold;
+        bool too_far_away;
+        double distance;
+    } accumulator{extractor, node_based_graph, threshold};
+
+    struct /*TrafficSignalBarrierRoadSelector*/
+    {
+        boost::optional<EdgeID> operator()(const NodeID,
+                                           const EdgeID,
+                                           const Intersection &intersection,
+                                           const util::NodeBasedDynamicGraph &) const
+        {
+            if (intersection.size() == 2)
+                return boost::make_optional(intersection[1].eid);
+            else
+                return boost::none;
+        }
+    } const selector;
+
+    (void)graph_walker.TraverseRoad(start, onto, accumulator, selector);
+
+    return accumulator.too_far_away;
+}
+
+bool SliproadHandler::isThroughStreet(const EdgeID from, const Intersection &intersection) const
```

Comment:
don't we have such a check somewhere else already? If not, I'd move it to the BaseClass